src/local_sensitivity/derivative_wrappers.jl

```diff
@@ -224,9 +224,20 @@ function _vecjacobian!(dλ, λ, p, t, S::SensitivityFunction, isautojacvec::Zygo
     else
       dλ[:] .= vec(tmp1)
     end
-
-    dgrad !== nothing && (dgrad[:] .= vec(tmp2))
     dy !== nothing && (dy[:] .= vec(_dy))
+    if dgrad !== nothing
```

Comment:
let's do this for now, but I think what we'll need to do is essentially revert it and in the SDE pass `dgrad = nothing`, and do this part separately. It can be much more efficient to calculate the Jacobian via forward mode in many cases, so we probably want options for how to build this Jacobian rather than hardcoding it. Then, that would keep the focus of `vecjacobian!` just on vector-Jacobian products. Since this loop doesn't share code with what's above it, it just doesn't make all that much sense in this spot.

Reply:
Let's open an issue to track this.